spec.md

```diff
@@ -690,6 +704,26 @@ message PublishVolumeInfo {
   // possible. This field is OPTIONAL.
   map<string, string> values = 1;
 }
+
+message FilesystemUser {
+  // The user name to be used for file access in this volume mount
+  // instance. This field is REQUIRED
+  string user_name = 1;
+
+  // The group name(s) to be used for file access through this volume
+  // mount instance This field is OPTIONAL
+  repeated string group_name = 2;
+
+  // The namespace that the user and groups belong to. This allows using
+  // the same user and/or group names in different namespaces.
+  // This field is OPTIONAL, if not provided the SPs default namespace
+  // is used.
+  string namespace = 3;
+
+  // Credentials to authenticate this mapping request.
+  // This field is OPTIONAL.
+  Credentials mapping_credentials = 4;
```

Comment:
I'm struggling to understand the need for this field. Are these the credentials of the user? There's already a credentials field for the RPC itself, why do we need even more here? Furthermore, a FilesystemUser object isn't a 
"mapping request", as I understand it. To me, FilesystemUser is an identity declaration.

Can you provide a short workflow/illustration that explains how this new credentials field would be used by a plugin?

Reply:
Adding this was a request from the last CIS sync conf call. People requested this to be added in order to authenticate the mapping request, personally I also think this could be covered by the existing credentials map but I wanted to follow up on the request.
The FilesystemUser object, if set,  defines which identity all file system access requests through this mount is mapped to. This prevents e.g. using the user identity "root:root" inside the container to access files on the mounted file system as user root. The ID "root:root" is mapped to the given FilesystemUser in the backend during rwx access.

The idea of the new credentials field is to provide credentials for the authentication of the mapping setup with the backend. Currently I'm not aware of CO / backend supporting this specific authentication.